packages/nimble-components/src/table/specs/table-columns-hld.md

```diff
@@ -109,43 +118,62 @@ abstract class TableColumn<TColumnConfig = {}> {
     @attr({ attribute: 'sort-direction' })
     public sortDirection: TableColumnSortDirection = TableColumnSortDirection.none;
 
-    // The relevant, static configuration a column requires its cell view to have access to.
-    columnConfig?: TColumnConfig;
+    // @internal Configuration settings for column plugin authors
+    public readonly columnInternals: ColumnInternals<TColumnConfig>;
+
+    public constructor(options: ColumnInternalsOptions) {
+        super();
+        this.columnInternals = new ColumnInternals(options);
+    }
+}
+```
+
+_Note: The `TableColumn` class may be updated to support other features not covered in this HLD such as grouping._
+
+### Column author internal configuration
+
+Column authors have a required `ColumnInternalsOptions` constructor parameter argument to define for static configuration and a `columnInternals` object that can be manipulated for dynamic configuration at runtime.
 
+```TS
+export interface ColumnInternalsOptions {
     // The tag (element name) of the custom element that renders the cell content for the column.
     // Should derive from TableCellView<TCellRecord, TColumnConfig>.
-    cellViewTag: string;
+    readonly cellViewTag: string;
 
     // The names of the fields that should be present in TCellRecord.
     // This array is parallel with the field names specified by `dataRecordFieldNames`.
-    readonly cellRecordFieldNames: readonly string[];
+    readonly cellRecordFieldNames: readonly TableFieldName[];
+}
+```
+
+```TS
+export class ColumnInternals<TColumnConfig> {
+    // The relevant, static configuration a column requires its cell view to have access to.
+    @observable
+    public columnConfig?: TColumnConfig;
 
     // The names of the fields from the row's record that correlate to the data that will be in TCellRecord.
     // This array is parallel with the field names specified by `cellRecordFieldNames`.
-    dataRecordFieldNames: readonly (TableFieldName | undefined)[] = [];
+    @observable
+    public dataRecordFieldNames: readonly (TableFieldName | undefined)[] = [];
 
     // The name of the data field that will be used for operations on the table, such as sorting and grouping.
-    operandDataRecordFieldName?: TableFieldName;
+    @observable
+    public operandDataRecordFieldName?: TableFieldName;
 
-    /**
-     * @internal
-     *
-     * The operation to use when sorting the table by this column.
-     */
+    // The operation to use when sorting the table by this column.
     @observable
     public sortOperation: TableColumnSortOperation;
 
-    // Function that allows the table column to validate the type that gets created
-    // for the cell data. This should validate that the types in TCellRecord are correct
-    // for each key defined by `cellRecordFieldNames`.
-    // This function should throw if validation fails.
-    abstract validateCellData(cellData: TCellRecord): void;
```

Comment:
@atmgrifter00 I removed this part from the spec since we haven't implemented it. The original TableColumn PR discussed adding this in a follow-up but we didn't track it.

I still think there may be a use-case, for example to be more robust for non-strict TypeScript users or for un-typed JS users, but we can revisit that later if it seems valuable.